src/connection.go

```diff
@@ -45,19 +35,19 @@ func (c configConnectionError) Error() string {
 	return "can't execute redis 'CONFIG' command: " + c.cause.Error()
 }
 
-func newSocketRedisCon(unixSocket string, options ...redis.DialOption) (conn, error) {
+func newSocketRedisCon(unixSocket string, options ...redis.DialOption) (redisConn, error) {
```

Comment:
why not returning a pointer to redisConn and a nil in case of error? isn't that cleaner?

Reply:
I think I made it return it the type rather than the pointer for performance reasons, but probably it won't be noticeable anywhere and it might cause some confusion. Let me fix that :)